### PR TITLE
CI matrix: Update JRuby versions

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR changes the CI matrix:

- remove the unused `sudo: false` Travis CI setting
- ~change the `rvm` CI matrix settings to use "known versions", with a reference to rvm's implementation: https://github.com/rvm/rvm/blob/master/config/known~
- Update JRuby version numbers

~The benefit is that changes like the one I proposed would be more automatic.~

### CODE REVIEW

**Comment by olleolleolle** [comment]
@JonRowe I tried to get this to use the latest JRuby, but the perhaps the `rvm` installation is OLD, and therefore not using latest-and-greatest.

Let's see if there's a way of updating it.

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 language: ruby
 script: "script/test_all"
-sudo: false
```
Comment (JonRowe) [inline .travis.yml]: Why was this necessary?

Comment (olleolleolle) [reply .travis.yml]: https://changelog.travis-ci.com/the-container-based-build-environment-is-fully-deprecated-84517

It's a change in Travis CI.

**Comment by JonRowe** [comment]
Thanks

**Comment by olleolleolle** [comment]
Thank _you_, maintaining RSpec & its related libraries is 🌳 important!
